modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/SynonymTokenFilterFactory.java

```diff
@@ -30,9 +30,57 @@
 import java.io.Reader;
 import java.io.StringReader;
 import java.util.List;
+import java.util.function.BiFunction;
 import java.util.function.Function;
 
 public class SynonymTokenFilterFactory extends AbstractTokenFilterFactory {
+    protected enum SynonymsSource {
+        INLINE("synonyms", SynonymTokenFilterFactory::getRulesReader_Inline),
```

Comment:
Should we have the standard camel case function names:

`getRulesReader_Inline` -> `getRulesReaderInline`?